### ISSUE TITLE
Update "breaks" SF addon

### ISSUE THREAD

**Issue by cg00001** [jmooremcc/plugin.program.super.favourites#9]
File missing:
![SF](https://github.com/jmooremcc/plugin.program.super.favourites/assets/38960370/faec995a-6f79-4d37-884d-21847b5f9f31)

**Comment by nebulous42069** [comment]
Seems it has to do with the location of Superfavourites folder in addons settings. Its blank on fresh install. Works if you set a path.. Doesnt work with special path.

**Comment by cg00001** [comment]
> Seems it has to do with the location of Superfavourites folder in addons settings. Its blank on fresh install. Works if you set a path.. Doesnt work with special path.

Right on the spot, it was:
`<setting id="FOLDER" default="true" />`

I tried a special, and didn't work:
`<setting id="FOLDER">special://home\userdata\addon_data\plugin.program.super.favourites\</setting>`

Then a direct link like you suggested, and it works fine:
`<setting id="FOLDER">/home/k/.kodi/userdata/addon_data/plugin.program.super.favourites/</setting>`

The easy way to fix until [jmooremcc](https://github.com/jmooremcc) fixes it and updates SF is from inside Kodi.
![sf](https://github.com/jmooremcc/plugin.program.super.favourites/assets/38960370/9226e400-1d07-4adb-84c0-32dbd1950263)

**Comment by jmooremcc** [comment]
> > Seems it has to do with the location of Superfavourites folder in addons settings. Its blank on fresh install. Works if you set a path.. Doesnt work with special path.
> 
> Right on the spot, it was: `<setting id="FOLDER" default="true" />`
> 
> I tried a special, and didn't work: `<setting id="FOLDER">special://home\userdata\addon_data\plugin.program.super.favourites\</setting>`
> 
> Then a direct link like you suggested, and it works fine: `<setting id="FOLDER">/home/k/.kodi/userdata/addon_data/plugin.program.super.favourites/</setting>`

Thanks for the feedback. 
I previously tested it on my Windows PC and Raspberry Pi and didn't encounter the problem. One of my testers ran into a similar problem and setting the folder location fixed the problem. 

I'm going to dig deeper into the issue. It should have read your existing settings file which already contains the location of your SF  folder.

**Comment by jmooremcc** [comment]
Setting the SF folder location in settings is the correct way to resolve the issue. In my testing, in which I cleaned out the entire SF folders, this error did not occur.

I will look into this and publish an update ASAP.

**Comment by nebulous42069** [comment]
> Setting the SF folder location in settings is the correct way to resolve the issue. In my testing, in which I cleaned out the entire SF folders, this error did not occur.
> 
> I will look into this and publish an update ASAP.
Try with a fresh install or delete the plugin.program.superfavorites folder in addon_data before installing SF to see the issue.  You will  get an error when trying to open SF. You have to manually add the SF location before it will work.

**Comment by nebulous42069** [comment]
>  In my testing, in which I cleaned out the entire SF folders, this error did not occur.
> 


Try with a fresh install or delete the plugin.program.superfavorites folder in addon_data before installing SF to see the issue. You will get an error when trying to open SF. You have to manually add the SF location before it will work.

**Comment by BMcGraw51** [comment]
These methods might work in Android 10 < but not in 11/12. Can't access the kodi data folder on a TV Box.

**Comment by cg00001** [comment]
> These methods might work in Android 10 < but not in 11/12. Can't access the kodi data folder on a TV Box.

Try:
> On the shield go to settings, Apps, Kodi, Permissions, Files and media , then click Allow all the time, you should now see zips so you can install addons and allow Kodi to access your external SD card or usb to add music, movies, and pictures.

**Comment by BMcGraw51** [comment]
I can access zips and media from anywhere on my network. The problem is I can't delete the super favourites folder or manage any of the files in the data folder. Write permission for any file manager fails.  Thanks anyway...

**Comment by nebulous42069** [comment]
> I can access zips and media from anywhere on my network. The problem is I can't delete the super favourites folder or manage any of the files in the data folder. Write permission for any file manager fails. Thanks anyway...

You can always use the file manager built into Kodi to access userdata
<img width="940" alt="Screenshot_5" src="https://github.com/jmooremcc/plugin.program.super.favourites/assets/125807561/96e37137-757f-4b2c-92ae-00de1b8c08f0">

**Comment by BMcGraw51** [comment]
You got me there! Didn't know about that feature. Bad news is that even though the sf folder is gone, I still can't edit the path. I click on the data folder from the sf settings and nothing happens - locked out.

**Comment by BMcGraw51** [comment]
Rolling back to v2.2.5 seems to be an acceptable alternative (until 2.3.2 is patched anyway)

**Comment by jmooremcc** [comment]
I've just uploaded SF version 2.3.3 which resolves the reported issue.

**Comment by BMcGraw51** [comment]
"THANK YOU SIR!" Much appreciated...

**Comment by nebulous42069** [comment]
Just curious, because I was able to fix this, but not sure if i had any other side effects. I deleted line 198 in default.py and it seemed to fix the issue. Is this the same fix you did, just curious because I was trying t figure it out for the last couple hours, lol.
<img width="621" alt="Screenshot_6" src="https://github.com/jmooremcc/plugin.program.super.favourites/assets/125807561/8e6189e0-79a8-4856-ad4f-241b721f484a">

**Comment by nebulous42069** [comment]
2.3.3 did not fix the issue for me. Same error. For now ill stick to my fix and just delete line 198..

**Comment by nebulous42069** [comment]
Update: Ok, it only works if you delete the addon_data and reinstall SF. It doesnt work and still errors if previous settings are in addon_data.

**Comment by jmooremcc** [comment]
> 

Can you post a link to your log?
Be sure to turn debugging on in SF settings.

**Comment by nebulous42069** [comment]
Ok, so had to delete the path i set in "Folder" setting and the path in "Default image" in settings.xml before it worked.

**Comment by nebulous42069** [comment]
Its fine, I can redo the settings file. It just wont update and work without deleting the settings.xml first. But if your curious heres the settings.xml that was working fine before the update, just doesnt work on the latest version for whatever reason. If you place this file in addon_data and run SF it errors, but not a big deal.. 
[settings.zip](https://github.com/jmooremcc/plugin.program.super.favourites/files/12307878/settings.zip)

**Comment by nebulous42069** [comment]
Ive always edited it for special paths so I can copy my setup to other devices, but seems the latest version doesnt like special://

**Comment by nebulous42069** [comment]
It seems like it just doesnt like the special path I had set for folder. All else is fine. Dont need it anyways as it should be default. It just wouldnt update the settings.xml with that path in it previously, no biggie.

**Comment by nebulous42069** [comment]
Ill just have to delete the settings.xml on my setups before updating to the latest version to fix it.

**Comment by cg00001** [comment]
I backed up my Super Favourites folder, and deleted SF from both addons and addon_data. Restarted, went to a SF shortcut , it installed and gave a popup. I placed my Super Favourites folder back in place, restarted and all works fine.
I think the issue would be if one has an earlier version and it auto-updates once transferred to another device.

BTW, there's a special working in settings.xml:
`<setting id="DOWNLOAD_FOLDER" default="true">special://profile/addon_data/plugin.program.super.favourites/downloads</setting>`

**Comment by nebulous42069** [comment]
I think it just fails to update to the new settings.xml format if previous settings.xml is present. Think thats what it comes down to..

**Comment by jmooremcc** [comment]
> Just curious, because I was able to fix this, but not sure if i had any other side effects. I deleted line 198 in default.py and it seemed to fix the issue. Is this the same fix you did, just curious because I was trying t figure it out for the last couple hours, lol. <img alt="Screenshot_6" width="621" src="https://user-images.githubusercontent.com/125807561/259579222-8e6189e0-79a8-4856-ad4f-241b721f484a.png">

Deleting that function call will affect the saving of icons & fanart when performing export/import operations.
Not a good idea to do that.
The fix I performed was to translate the special file for the ROOT definition in util.py.

**Comment by DJMix-I-Can** [comment]
Thought I'd chime in for others. I have four Nvidia Shields with external USB sticks. I have my Super Favourites folder saved to my USB sticks. Personally, I waited to see if there was going to be an issue with the update & then the comments started coming in. I didn't update on three of them. I just upped the version number. If it's not broken why update it. Unfortunately, on one of the 2019 models, I clicked on update all. Super Favorites would not open any of my folders even though the path of the location is still the same. The only thing I did was go into settings, general, location of super favorites folder. The window will open up to set a path, back out, back out, back out, back out until you get to the screen where you get the options of which location to store it to. Ex: external storage, internal, mapped drives, NFS, root filesystem etc. etc.. Once you reach that menu, redo the path you originally had it to & click OK.  That should resolve your issue.
